addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/channelhelper/RadioHelper.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.netatmo.internal.channelhelper;
+
+import static org.openhab.binding.netatmo.NetatmoBindingConstants.*;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link RadioHelper} handle specific behavior
+ * of WIFI or RF devices and modules
+ *
+ * @author Gaël L'hopital - Initial contribution
+ *
+ */
+public class RadioHelper {
+    private static Logger logger = LoggerFactory.getLogger(RadioHelper.class);
+    final private List<Integer> signalThresholds = new ArrayList<>();
+    Object module;
+
+    public RadioHelper(String signalLevels) {
+        List<String> thresholds = Arrays.asList(signalLevels.split(","));
+        thresholds.forEach(t -> signalThresholds.add(Integer.parseInt(t)));
+    }
+
+    private int getSignalStrength(int signalLevel) {
+        int level;
```

Comment:
If we are on the Java 8 tour, this could work ;-)

```
signalThresholds.filter(s -> s > level).findFirst().orElse(Collections.max(signalThresholds));
```

BTW: do you guarantee somewhere that those lists of values are ordered otherwise the code makes some invalid assumptions here.